specifyweb/frontend/js_src/lib/components/WorkBench/WbView.tsx

```diff
@@ -49,13 +48,298 @@ import type { WbMapping } from './mapping';
+        colsToInclude.add(physicalCol);
+      })
+    );
+    const rowsToHide = workbench.data
+      .map((_, physicalRow) => physicalRow)
+      .filter(
+        (physicalRow) =>
+          !rowsToInclude.has(physicalRow) &&
+          !initialHiddenRows.includes(physicalRow)
+      )
+      .map(workbench.hot.toVisualRow);
+    const colsToHide = workbench.dataset.columns
+      .map((_, physicalCol) => physicalCol)
+      .filter(
+        (physicalCol) =>
+          !colsToInclude.has(physicalCol) &&
+          !initialHiddenCols.includes(physicalCol)
+      )
+      .map(workbench.hot.toVisualColumn);
+
+    if (showResults) {
+      workbench.hot?.updateSettings({ readOnly: true });
+      getHotPlugin(workbench.hot, 'hiddenRows').hideRows(rowsToHide);
+      getHotPlugin(workbench.hot, 'hiddenColumns').hideColumns(colsToHide);
+
+      workbench.utils.toggleCellTypes(
+        'newCells',
+        'remove',
+        spreadsheetContainer.current
+      );
+    } else {
+      getHotPlugin(workbench.hot, 'hiddenRows').showRows(
+        rowsToHide.filter((visualRow) => !initialHiddenRows.includes(visualRow))
+      );
+      getHotPlugin(workbench.hot, 'hiddenColumns').showColumns(
+        colsToHide.filter((visualCol) => !initialHiddenCols.includes(visualCol))
+      );
+    }
+
+    workbench.hot.render();
+  }, [showResults]);
+
+  return (
+    <>
+      <div
+        className="flex items-center justify-between gap-x-1 gap-y-2 whitespace-nowrap"
+        role="toolbar"
+      >
+        <div className="contents">
+          <DataSetName
+            dataset={dataset}
+            getRowCount={() =>
+              hotRef.current?.hotInstance
+                ? dataset.rows.length
+                : hotRef.current!.hotInstance!.countRows() -
+                  hotRef.current!.hotInstance!.countEmptyRows(true)
+            }
+          />
+        </div>
+        <Button.Small
+          aria-haspopup="grid"
+          aria-pressed={showToolkit}
+          onClick={toggleToolkit}
+        >
+          {commonText.tools()}
+        </Button.Small>
+        <span className="-ml-1 flex-1" />
+        {canUpdate || isMapped ? (
+          <Link.Small href={`/specify/workbench/plan/${dataset.id}/`}>
+            {wbPlanText.dataMapper()}
+          </Link.Small>
+        ) : undefined}
+        <WbActionsComponent
+          dataset={dataset}
+          hasUnsavedChanges={hasUnsavedChanges}
+          isUploaded={isUploaded}
+          onRefresh={triggerRefresh}
+          mappings={mappings!}
+          checkDeletedFail={checkDeletedFail}
+          onSpreadsheetUpToDate={spreadsheetUpToDate}
+          workbench={workbench}
+          onToggleResults={toggleResults}
+        />
+      </div>
+      {showToolkit ? (
+        <WbToolkit
+          dataset={dataset}
+          hotRef={hotRef}
+          mappings={mappings!}
+          data={data}
+          handleDatasetDelete={handleDatasetDelete}
```

Comment:
this callback doesn't actually delete the dataset but only signals the parent that the dataset is now deleted and that a corresponding dialog should be displayed to the user, right?
if so, a better name would be `handleDatasetDeleted` rather than `handleDatasetDelete` to make it sure that this callback is not deleting a dataset, but reacting to dataset being deleted